### PR TITLE
[cloud-sql-proxy] Add support for Rocky and CentOS

### PR DESCRIPTION
**Pull request by functicons**
Debian/Ubuntu and Rocky/CentOS have different MySQL config dir and systemd service names.

### CODE REVIEW

File: cloud-sql-proxy/cloud-sql-proxy.sh [hunk 1]
```diff
@@ -314,8 +314,16 @@ EOF
...
+  if [[ -d /etc/mysql/conf.d ]]; then
+    mysql_conf_dir=/etc/mysql/conf.d
+  elif [[ -d /etc/my.cnf.d ]]; then
+    mysql_conf_dir=/etc/my.cnf.d
+  else
+    err "Neither /etc/mysql/conf.d nor /etc/my.cnf.d exists"
+  fi
   # Configure MySQL client to talk to metastore
-  cat <<EOF >/etc/mysql/conf.d/cloud-sql-proxy.cnf
+  cat <<EOF >"${mysql_conf_dir}/cloud-sql-proxy.cnf"
```
Comment (medb) [inline cloud-sql-proxy/cloud-sql-proxy.sh:326]: Please uncomment test on Rocky to make sure that change works: https://github.com/GoogleCloudDataproc/initialization-actions/blob/master/cloud-sql-proxy/test_cloud_sql_proxy.py#L59-L60

**Review by medb — CHANGES_REQUESTED** [review]

File: cloud-sql-proxy/cloud-sql-proxy.sh (hunk 1 above)
Comment (functicons) [reply cloud-sql-proxy/cloud-sql-proxy.sh:326]: Done.

File: cloud-sql-proxy/test_cloud_sql_proxy.py [hunk 2]
```diff
@@ -56,9 +56,6 @@ def __submit_pyspark_job(self, cluster_name):
       'HA',
   )
   def test_cloud_sql_proxy(self, configuration):
```
Comment (medb) [inline cloud-sql-proxy/test_cloud_sql_proxy.py:58]: Seems like this test is failing during presubmit, may you fix it?

**Review by medb — CHANGES_REQUESTED** [review]

File: cloud-sql-proxy/test_cloud_sql_proxy.py (hunk 2 above)
Comment (medb) [reply cloud-sql-proxy/test_cloud_sql_proxy.py:58]: Seems like HA test is still failing on Debian 10 images.

**Review by medb — CHANGES_REQUESTED** [review]

File: cloud-sql-proxy/test_cloud_sql_proxy.py (hunk 2 above)
Comment (functicons) [reply cloud-sql-proxy/test_cloud_sql_proxy.py:58]: I couldn't reproduce the error. Maybe flaky? I'm rerunning the test.

Comment (medb) [reply cloud-sql-proxy/test_cloud_sql_proxy.py:58]: Seems like it consistently fails in presubmit

**Review by medb — APPROVED** [review]

**Pull request merged**